src/components/vertical-menu/responsive-vertical-menu/responsive-vertical-menu.component.tsx

```diff
@@ -0,0 +1,266 @@
+  const reduceMotion = !useMediaQuery(
+    "screen and (prefers-reduced-motion: no-preference)",
+  );
+
+  const { current: menu } = menuRef;
+  const { current: button } = buttonRef;
+
+  const measureDimensions = useCallback(() => {
+    // Only calculate the width of the menu if the menu is open in responsive mode
+    if (active && responsiveMode) {
+      // Get the width of the menu container. The ref might not be set yet, hence
+      // the use of `document.querySelector` to find the menu container.
+      const menuContainer =
+        menu ||
+        document.querySelector("[id='responsive-vertical-menu-primary']");
+      /* istanbul ignore else */
+      if (menuContainer) {
+        setResponsiveWidth(
+          `${menuContainer.getBoundingClientRect().width - 16}px`,
+        );
+      }
+    }
+
+    if (activeMenuItem && menu) {
+      setLeft(`${menu.getBoundingClientRect().right}px`);
+    } else {
+      setLeft("auto");
+    }
+
+    if (activeMenuItem && button) {
+      setTop(`${button.getBoundingClientRect().bottom}px`);
+    } else {
+      setTop("auto");
+    }
+  }, [active, menu, responsiveMode, activeMenuItem, button]);
+
+  const handleOutsideClick = useCallback(
+    (event: MouseEvent) => {
+      const notInContainer =
+        containerRef.current &&
+        !containerRef.current.contains(event.target as Node);
+
+      if (notInContainer) {
+        setActive(false);
+      }
+    },
+    [containerRef],
+  );
+
+  const handleActiveToggle = useCallback(() => {
+    setActive((previousState) => !previousState);
+    // Make sure the menu is closed when the button is clicked (prevents historic menu items being retained in memory)
+    setActiveMenuItem(null);
+  }, [setActiveMenuItem]);
+
+  useLayoutEffect(() => {
+    measureDimensions();
+  }, [active, measureDimensions, menu, responsiveMode]);
+
+  useEffect(() => {
+    const handleBlur = () => {
+      setTimeout(() => {
+        if (
+          containerRef.current &&
+          !containerRef.current.contains(document.activeElement)
+        ) {
+          setActive(false);
+        }
+      }, 0);
+    };
+
+    const handleClose = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        e.preventDefault();
+
+        setActive(false);
+      }
+    };
+
+    const currentContainer = containerRef.current;
+
+    if (active && !responsiveMode) {
+      document.addEventListener("keydown", handleClose);
+      window.addEventListener("click", handleOutsideClick);
+      currentContainer?.addEventListener("focusout", handleBlur);
+    }
+
+    return () => {
+      document.removeEventListener("keydown", handleClose);
+      window.removeEventListener("click", handleOutsideClick);
+      currentContainer?.removeEventListener("focusout", handleBlur);
+    };
+  }, [active, containerRef, handleOutsideClick, responsiveMode]);
+
+  useEffect(() => {
+    setReducedMotion?.(reduceMotion);
+    setResponsiveMode?.(!largeScreen || false);
```

Comment:
question: won't this just be false || false on some occasions? If so we could just go with the following unless I'm missing something

```
setResponsiveMode?.(!largeScreen);